.github/workflows/build-images-base.yaml

```diff
@@ -23,6 +23,7 @@ concurrency:
 
 jobs:
   build-and-push:
+    name: Build and Push Images
```

Comment:
Do you think it's useful to have names on all jobs? I'd advocate for explicitly not setting names on jobs unless absolutely useful, and instead make sure the job IDs correctly represent what the job is doing. I think in general the less we need to worry about additional fields the better.

Reply:
The idea behind giving these names is to make them more readable for new report tooling

Reply:
But is there really a big readability advantage in having jobs named `Build and Push Images` instead of `build-and-push`, as long as they have the same name everywhere? 🤔 I would advocate for not introducing bloat if we can avoid it, if everybody else is convinced this is sound then let's do it but I personally have no problem with having `build-and-push` as job name in the dashboard.

Reply:
I am not sure how much of a readability advantage. I have added a workflow with both jobs with names and without to see the difference. In my option, it is more present if the names are separated by space.

![Screenshot](https://github.com/cilium/cilium/assets/3773321/4fb9cc4e-dc66-44e5-8be3-f310f8e840c8)

Reply:
I guess the question is "is it pleasant enough to warrant additional bloat in workflows". Personally I would say no, but again I only one person, would be nice to have more opinions :D

Reply:
@brlbil I don't understand your comment, should there be two pictures?

Reply:
Although 'name' is not mandatory, we should still add it to make it explicit and consistent across other workflows.

Reply:
Consistency I agree with -- but if there's already the same ID across workflows, then I'm personally not in favor of additionally copy/pasting the same name. Again, if I'm alone on this hill, let's just go with it :D